result_test.go

```diff
@@ -409,11 +409,13 @@ func TestToMapBy(t *testing.T) {
 
 func TestToSlice(t *testing.T) {
```

Comment:
> len(out)>0 and appending/growing doesn't lose the original elements in out.
@ahmetalpbalkan the behavior in this scenario will be: copy the result reusing the len(out) and appending if len(result) > len(out). In this case output slice will be overwritten(and appended if necessary) with the result elements. If len(out) > len(result) the elements of result will be copied to the output and resized to len(result). What do you think?